swap_meet/vendor.py

```diff
@@ -1,2 +1,156 @@
+from curses import beep
+from nis import cat
+from swap_meet.item import Item
+from swap_meet.clothing import Clothing
+from swap_meet.decor import Decor
+from swap_meet.electronics import Electronics
+
 class Vendor:
-    pass
\ No newline at end of file
+    """Vendor has a relationship with Item class called composition."""
+    
+
+    # Wave 1
+    def __init__(self, inventory = None):
+        """Inventory is keyword argument that optionally pass in."""
+        
+        # assign empty list to inventory when the if statement is falsy. 
+        # Otherwise, assign inventory as value of object's inventory.
+        if not inventory:
+            inventory = []
+        self.inventory = inventory
+        
+        
+    def add(self, item):
+        """Adding new item into inventory and returns that item."""
+        
+        # make sure input is not empty before adding
+        if item:
+            self.inventory.append(item)
+            return item
+        
+
+    def remove(self, item):
+        """Removing item from inventory if it is matching requirement and returns that removed item."""
+        
+        # make sure input is not empty before removing
+        if item and item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+
+    # Wave 2
+    def get_by_category(self, category):
+        """
+        - Adding item in a list if it matches category then return that list of item,
+        - Otherwise, return []
+        """
+        
+        items_list = []
+        # adding item into the new list if the inventory is not empty and matches category
+        if category and len(self.inventory) > 0:
+            for item in self.inventory:
+                if item.category == category:
+                    items_list.append(item)
+            return items_list
+        return []
```

Comment:
We might not want to check whether category is truthy here. Maybe it is meaningful to have an item with an empty string for a category! And even if we decide that's invalid, it would probably be better to raise an exception than silently returning an empty list.